### PR TITLE
W3C / HTML5 fix

### PR DESCRIPTION
**Pull request by AnaelMobilia**
As a continuation of #1202 & PR #1206 : fix some errors & warnings on the W3C validator

### CODE REVIEW

File: inc/toolbox.class.php [hunk 1]
```diff
@@ -785,16 +785,15 @@ static function stripslashes_deep($value) {
    **/
    static function append_params($array, $separator='&', $parent='') {
 
-      $params = array();
+        $params = array();
```
Comment (trasher) [inline inc/toolbox.class.php]: Indentation "problem", test should have fail :/

File: inc/toolbox.class.php [hunk 2]
```diff
@@ -785,16 +785,15 @@ static function stripslashes_deep($value) {
...
       foreach ($array as $k => $v) {
 
          if (is_array($v)) {
             $params[] = self::append_params($v, $separator,
                                             (empty($parent) ? rawurlencode($k)
-                                                            : $parent .'['.rawurlencode($k).']'));
+                                                            : $parent . '%5B' . rawurlencode($k) . '%5D'));
```
Comment (trasher) [inline inc/toolbox.class.php]: Hum... I'm not sure about this one... Why did you replace it? I do not get the point regarding to HTML validation :)

File: inc/toolbox.class.php [hunk 3]
```diff
@@ -785,16 +785,15 @@ static function stripslashes_deep($value) {
...
          } else {
-            $params[] = (!empty($parent) ? $parent . '[' . rawurlencode($k) . ']'
-                                         : rawurlencode($k)) . '=' . rawurlencode($v);
+            $params[] = (!empty($parent) ? $parent . '%5B' . rawurlencode($k) . '%5D' : rawurlencode($k)) . '=' . rawurlencode($v);
```
Comment (trasher) [inline inc/toolbox.class.php]: Same as the above

File: inc/toolbox.class.php (hunk 1 above)
Comment (AnaelMobilia) [inline inc/toolbox.class.php]: Nice catch ! :-)

File: inc/toolbox.class.php (hunk 2 above)
Comment (AnaelMobilia) [inline inc/toolbox.class.php]: I get an W3C error on bad URL formating.
I find it on the ticket view.
The button image have the search parameters on the href of the a element.

Without this patch, we provide an element like 
```
<a href="/glpi/front/ticket.php188?reset=reset&amp;criteria[0][field]=55&amp;criteria[0][searchtype]=equals&amp;criteria[0][value]=2&amp;criteria[0][link]=AND&amp;criteria[1][field]=59&amp;criteria[1][searchtype]=equals&amp;criteria[1][value]=2&amp;criteria[1][link]=AND"><img title="Ticket attendant votre validation" alt="Ticket attendant votre validation" src="/glpi/pics/menu_showall.png" class="pointer"></a>
```
This code will throw an error on W3C validator as "[" and "]" elements aren't allowed in a URL

With this patch, we provide an element like 
```
<a href="/glpi/front/ticket.php188?reset=reset&amp;criteria%5B0%5D%5Bfield%5D=55&amp;criteria%5B0%5D%5Bsearchtype%5D=equals&amp;criteria%5B0%5D%5Bvalue%5D=2&amp;criteria%5B0%5D%5Blink%5D=AND&amp;criteria%5B1%5D%5Bfield%5D=59&amp;criteria%5B1%5D%5Bsearchtype%5D=equals&amp;criteria%5B1%5D%5Bvalue%5D=2&amp;criteria%5B1%5D%5Blink%5D=AND"><img title="Ticket attendant votre validation" alt="Ticket attendant votre validation" src="/glpi/pics/menu_showall.png" class="pointer"></a>
```

(same for above)

**Comment by AnaelMobilia** [comment]
@trasher 
Thanks for reviewing !
I answered to your remarks

Regards

File: inc/toolbox.class.php (hunk 2 above)
Comment (trasher) [inline inc/toolbox.class.php]: OK, I see.

The `[]` notation is used on form submission with array of values; if the feature still works as expected; than there is no problem with that change :)

**Comment by trasher** [comment]
You're welcome ;)

But now, I'm about to go back home; I'll finish the review next year :D

File: inc/toolbox.class.php (hunk 2 above)
Comment (AnaelMobilia) [inline inc/toolbox.class.php]: As I tested, no problems occured. (PHP 5.5)
It's the browser (nor the server ?) job to "translate" ;-)

**Comment by AnaelMobilia** [comment]
Good end of year :)

**Comment by trasher** [comment]
Great job, thank you :)

**Pull request merged**